modules/canbus/vehicle/lincoln/lincoln_controller.cc

```diff
@@ -342,16 +342,19 @@ Chassis LincolnController::chassis() {
   }
 
   // give engage_advice based on error_code and canbus feedback
-  if (!chassis_error_mask_ && !chassis_.parking_brake() &&
-      (chassis_.throttle_percentage() != 0.0) &&
-      (chassis_.brake_percentage() != 0.0)) {
+  if (chassis_error_mask_ || (chassis_.throttle_percentage() == 0.0) ||
+      (chassis_.brake_percentage() == 0.0)) {
     chassis_.mutable_engage_advice()->set_advice(
-        apollo::common::EngageAdvice::READY_TO_ENGAGE);
-  } else {
+        apollo::common::EngageAdvice::DISALLOW_ENGAGE);
+    chassis_.mutable_engage_advice()->set_reason("Chassis error!");
+  } else if (chassis_.parking_brake() || !CheckSafetyError(chassis_detail)) {
     chassis_.mutable_engage_advice()->set_advice(
         apollo::common::EngageAdvice::DISALLOW_ENGAGE);
     chassis_.mutable_engage_advice()->set_reason(
-        "CANBUS not ready, firmware error or emergency button pressed!");
+        "Vehicle is not in a safe state to engage!");
```

Comment:
the reason needs to be more specific. Otherwise, the driver or passenger would not know why not allowed to engage.